src/Build/Graph/ProjectInterpretation.cs

```diff
@@ -227,6 +232,16 @@ ProjectItemInstance projectReference
             // The properties on the project reference supersede the ones from the MSBuild task instead of appending.
             if (newProperties.Count == 0)
             {
+                if(ConversionUtilities.ValidBooleanTrue(projectReference.Project.GetPropertyValue(EnableDynamicPlatformResolutionMetadataName)) && String.IsNullOrEmpty(projectReference.GetMetadataValue(SetPlatformMetadataName)))
```

Comment:
Add a comment which calls out that it's mimicking the target, so that it's there for reference